probe.c

```diff
@@ -169,13 +169,13 @@ static int is_openvpn_protocol (const char*p,ssize_t len, struct sslhcfg_protoco
          * whereas the packet id is increased with every transmitted datagram.
          */
 
-        if (len <= OVPN_HARD_RESET_PACKET_ID_OFFSET(OVPN_HMAC_128))
+        if (len <= OVPN_HARD_RESET_PACKET_ID_OFFSET(OVPN_HMAC_128) + 4)
             return PROBE_NEXT;
 
         if (ntohl(*(uint32_t*)(p + OVPN_HARD_RESET_PACKET_ID_OFFSET(OVPN_HMAC_128))) <= 5u)
```

Comment:
We need to make sure that the size of the buffer is at least `OVPN_HARD_RESET_PACKET_ID_OFFSET(OVPN_HMAC_128) + sizeof(uint32_t)` (9 + 16 + 4)  big.